### PR TITLE
feat: autosubmit at all digits to enable 2fa

### LINKED ISSUE #9726: [CAL-2031] auto submit 2FA if all inputs have a value
once these 6 have been filled, auto submit the form

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2031](https://linear.app/calcom/issue/CAL-2031/auto-submit-2fa-if-all-inputs-have-a-value)</sub>

### PR DESCRIPTION
**Pull request by sohamdata**
## What does this PR do?

Adds Auto-submit to enable 2FA.

Fixes #9726 

## Demo:
success

## Type of change

- New feature
## Tested the following:

- enable 2FA, check payload and response [200]
- disable 2FA, check payload and response [200]

**tl;dr changes**: added `optional?` params to `TwoFactor.tsx` to handle submit, using just **useEffect** to check for the length of 2fa code with change in the overall `value` of the code to prevent multiple post requests.

### Notes for maintainer:
- Spotted a bug in disabling 2FA, as you can see towards the end of the video above, 2 "Two-Factor Code" text appears:
image
Created an issue for it: #9761 

/claim #9726

### CODE REVIEW

**Comment by sohamdata** [comment]
@PeerRich any update?

File: apps/web/components/auth/TwoFactor.tsx [hunk 1]
```diff
@@ -5,7 +5,13 @@ import { useFormContext } from "react-hook-form";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { Label, Input } from "@calcom/ui";
 
-export default function TwoFactor({ center = true }) {
+interface TwoFactorProps {
+  center?: boolean;
+  onSubmit?: (values: any, e: any) => void;
```
Comment (PeerRich) [inline apps/web/components/auth/TwoFactor.tsx:10]: can we find a type here instead of any?

Comment (sohamdata) [reply apps/web/components/auth/TwoFactor.tsx]: added types: [bd7f62f](https://github.com/calcom/cal.com/pull/9759/commits/bd7f62fb84178ac5fbb37ada2c3c5c08ce92d3f3?diff=split&w=0)

**Comment by emrysal** [comment]
Hi @sohamdata thank you for this! I compared both your and @afzalsayed96's implementations and I'm leaning more to the latter. I believe you've seen that as well and would agree.

**Pull request closed without merging**